openshift/templates/app/app-digmkt-deploy.yaml

```diff
@@ -254,4 +254,9 @@ parameters:
 
 - name: BASIC_AUTH_USERNAME
   displayName: The username for basic auth protection on test environments.
-  required: false
\ No newline at end of file
+  required: false
+
+- name: ORIGIN
+  displayName: The vanity url for the deployment
+  value: https://${NAME}-${APP_GROUP}-${TAG_NAME}.apps.silver.devops.gov.bc.ca
+  required: true
```

Comment:
Why make the new template variable required when it has a good default value?

Reply:
> @thegentlemanphysicist have you tested these changes with a deployment?

I redeployed dev and test, though it's probably best someone else tries the deployment to make sure the modified readme still work @BryanGK I'll document the two cases on the PR's description

Reply:
@wenzowski I do not believe you can pass in params to a default param in these yaml config files (i tried).  Meaning the ${NAME}, ${APP_GROUP}, and ${TAG_NAME} are treated as strings when the template is processed.  Despite the fact that it's not a 'good' or 'valid' default value the form of the ORIGIN param, having it's for defined will allow us to easily fix the deployment config if the form of these urls ever change or if our pipeline structure changes (i.e. we no longer use a reverse proxy in test, or we have multiple named deployments in dev, etc)

All that said I agree it is a bit _hacky_ I'm not certain the templates are a flexible enough framework to allow something equivalent to a conditional statements to be used.   

ORIGIN: ${ORIGIN ? ORIGIN :  https://${NAME}-${APP_GROUP}-${TAG_NAME}.apps.silver.devops.gov.bc.ca }

Reply:
Interesting, so the default value parameter isn’t interpolated. That’s a reason to perhaps consider migrating to Helm from the default OpenShift templating system. I agree that an expressive default value like this helps understand what string to pass in to the template.